haystack_experimental/memory_stores/mem0/src/mem0/memory_store.py

```diff
@@ -0,0 +1,227 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Optional
+
+from haystack import default_from_dict, default_to_dict
+from haystack.dataclasses.chat_message import ChatMessage
+from haystack.lazy_imports import LazyImport
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+with LazyImport(message="Run 'pip install mem0ai'") as mem0_import:
+    from mem0 import Memory, MemoryClient
```

Comment:
We are adding `mem0` as a dependency to the project so no need to put it under a `LazyImport`. Long term I imagine we will make this a core-integration and the `Agent` will only use the to-be-made protocol which we can put into haystack main.